src/teachy_tv.c

```diff
@@ -0,0 +1,1775 @@
+extern void sub_8159F40();
+
+void TeachyTvCallback();
+void TeachyTvMainCallback();
+void TeachyTvVblankHandler();
+void TeachyTvCreateAndRenderRbox();
+void TeachyTvInitIo();
+u8 TeachyTvSetupObjEventAndOam();
+void TeachyTvSetupPostBattleWindowAndObj(u8);
+u8 TeachyTvSetupWindow();
+void TeachyTvSetupScrollIndicatorArrowPair();
+void TeachyTvSetWindowRegs();
+void TeachyTvSetupBg();
+void TeachyTvLoadGraphic();
+void TeachyTvPostBattleFadeControl(u8);
+void TeachyTvOptionListController(u8);
+void TeachyTvAudioByInput(s32, bool8, struct ListMenu *);
+void TeachyTvQuitFadeControlAndTaskDel(u8 taskId);
+void TeachyTvRenderMsgAndSwitchClusterFuncs(u8 taskId);
+void TeachyTvClearBg1EndGraphicText();
+void TeachyTvBackToOptionList(u8 taskId);
+void TeachyTvSetupBagItemsByOptionChosen();
+void TeachyTvPrepBattle(u8 taskId);
+void TeachyTvGrassAnimationMain(u8 taskId, s16 x, s16 y, u8 subpriority, bool8 mode);
+void TeachyTvLoadBg3Map(void *);
+u8 TeachyTvGrassAnimationCheckIfNeedsToGenerateGrassObj(s16 x, s16 y);
+void TeachyTvGrassAnimationObjCallback(struct Sprite *sprite);
+void TeachyTvRestorePlayerPartyCallback();
+void TeachyTvPreBattleAnimAndSetBattleCallback(u8 taskId);
+void TeachyTvLoadMapTilesetToBuffer(struct Tileset *ts, u8 *dstBuffer, u16 size);
+void TeachyTvPushBackNewMapPalIndexArrayEntry(struct MapData *mStruct, u16 *buf1, u8 *palIndexArray, u16 mapEntry, u16 offset);
+void TeachyTvComputeMapTilesFromTilesetAndMetaTiles(u16 *metaTilesArray, u8 *blockBuf, u8 *tileset);
+void TeachyTvComputeSingleMapTileBlockFromTilesetAndMetaTiles(u8 *blockBuf, u8 *tileset, u8 metaTile);
+u16 TeachyTvComputePalIndexArrayEntryByMetaTile(u8 *palIndexArrayBuf, u16 metaTile);
+void TeachyTvLoadMapPalette(const struct MapData *const mStruct, u8 *palIndexArray);
+
+void TeachyTvCallback()
+{
+    RunTasks();
+    AnimateSprites();
+    BuildOamBuffer();
+    do_scheduled_bg_tilemap_copies_to_vram();
+    UpdatePaletteFade();
+}
+
+void TeachyTvVblankHandler()
+{
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+}
+
+void sub_815ABC4(u8 mode, void (*cb)())
+{
+    gTeachyTV_StaticResources.mode = mode;
+    gTeachyTV_StaticResources.callback = cb;
+    if (mode == 0)
+    {
+        gTeachyTV_StaticResources.scrollOffset = 0;
+        gTeachyTV_StaticResources.selectedRow = 0;
+        gTeachyTV_StaticResources.optionChosen = 0;
+    }
+    if (mode == 1)
+    {
+        gTeachyTV_StaticResources.mode = 0;
+    }
+    SetMainCallback2(TeachyTvMainCallback);
+}
+
+void sub_815ABFC()
+{
+    if (gTeachyTV_StaticResources.mode == 1)
+        sub_815ABC4(1, gTeachyTV_StaticResources.callback);
+    else
+        sub_815ABC4(2, gTeachyTV_StaticResources.callback);
+}
+
+void sub_815AC20()
+{
+    gTeachyTV_StaticResources.mode = 1;
+}
+
+void TeachyTvMainCallback()
+{
+    int state;
+    int taskId;
+    struct Task *taskAddr;
+    struct TeachyTvBuf **memBuf;
+    u32 x;
+
+    state = gMain.state;
+    switch (state)
```

Comment:
`switch (gMain.state)`

Reply:
This one failed

Reply:
Did you try removing the unnecessary temporary variables taskAddr, state, and memBuf?